artifactory/commands/golang/archive.go

```diff
@@ -77,27 +73,16 @@ import (
 // Note that this package does not provide hashing functionality. See
 // golang.org/x/mod/sumdb/dirhash.
 
-const (
-	// MaxZipFile is the maximum size in bytes of a module zip file. The
-	// go command will report an error if either the zip file or its extracted
-	// content is larger than this.
-	MaxZipFile = 500 << 20
-
-	// MaxGoMod is the maximum size in bytes of a go.mod file within a
-	// module zip file.
-	MaxGoMod = 16 << 20
-
-	// MaxLICENSE is the maximum size in bytes of a LICENSE file within a
-	// module zip file.
-	MaxLICENSE = 16 << 20
-)
-
 // Archive project files according to the go project standard
-func archiveProject(writer io.Writer, dir, mod, version string) error {
+func archiveProject(writer io.Writer, dir, mod, version string, excludedPatterns []string) error {
 	m := module.Version{Version: version, Path: mod}
-	var files []File
-
-	err := filepath.Walk(dir, func(filePath string, info os.FileInfo, err error) error {
+	excludedPatterns, err := getAbsolutePaths(excludedPatterns)
+	if err != nil {
+		return err
+	}
+	excludePathPattern := fspatterns.PrepareExcludePathPattern(excludedPatterns, utils.GetPatternType(utils.PatternTypes{RegExp: false, Ant: false}), true)
```

Comment:
excludePathPattern --> excludePatternsStr
(this string can include multiple patterns, and not necessarily a single pattern)